plugins/services/src/js/containers/service-detail/ServiceDetail.js

```diff
@@ -187,50 +160,43 @@ class ServiceDetail extends mixin(TabsMixin) {
     const { service: { id } } = this.props;
     const routePrefix = `/services/detail/${encodeURIComponent(id)}`;
 
-    const tabs = [];
-    const activeTab = this.state.currentTab;
-
-    tabs.push({
-      label: "Instances",
-      callback: () => {
-        this.setState({ currentTab: "tasks" });
-      },
-      isActive: activeTab === "tasks"
-    });
-
-    tabs.push({
-      label: "Configuration",
-      callback: () => {
-        this.setState({ currentTab: "configuration" });
-      },
-      isActive: activeTab === "configuration"
-    });
-
-    tabs.push({
-      label: "Debug",
-      callback: () => {
-        this.setState({ currentTab: "debug" });
-      },
-      isActive: activeTab === "debug"
-    });
+    const tabs = [
+      { label: "Instances", routePath: `${routePrefix}/tasks` },
+      { label: "Configuration", routePath: `${routePrefix}/configuration` },
+      { label: "Debug", routePath: `${routePrefix}/debug` }
+    ];
 
     if (this.hasVolumes()) {
       tabs.push({
         label: "Volumes",
-        routePath: routePrefix + "/volumes",
-        callback: () => {
-          this.setState({ currentTab: "volumes" });
-        },
-        isActive: activeTab === "volumes"
+        routePath: `${routePrefix}/volumes`
       });
     }
 
     return tabs;
   }
 
   render() {
-    const { children, service: { id } } = this.props;
-    const breadcrumbs = <ServiceBreadcrumbs serviceID={id} />;
+    const { children, actions, errors, params, routes, service } = this.props;
+    const breadcrumbs = <ServiceBreadcrumbs serviceID={service.id} />;
+    const clonedProps = {
```

Comment:
I have already answered that before, but I'll answer it again;
Yes, we "need" to clone props since we are passing them down to their route.
It would be nice to get these as parameters in the route, but that requires us to change around logic in each individual tab, and in an effort to try to contain this change in as small chunks as possible, I figured this was the path of least resistance.

And I'll add a comment about it.